src/autowiring/CoreThreadLinux.cpp

```diff
@@ -23,3 +24,39 @@ void BasicThread::GetThreadTimes(std::chrono::milliseconds& kernelTime, std::chr
   kernelTime = std::chrono::duration_cast<milliseconds>(seconds(usage.ru_stime.tv_sec) + microseconds(usage.ru_stime.tv_usec));
   userTime = std::chrono::duration_cast<milliseconds>(seconds(usage.ru_utime.tv_sec) + microseconds(usage.ru_utime.tv_usec));
 }
+
+void BasicThread::SetThreadPriority(ThreadPriority threadPriority) {
+  struct sched_param param = { 0 };
+  int policy = SCHED_RR;
+  int percent = 0;
+
+  switch (threadPriority) {
+  case ThreadPriority::Idle:
+    policy = SCHED_IDLE;
+    percent = 0;
+    break;
+  case ThreadPriority::Lowest:
+    percent = 1;
+    break;
+  case ThreadPriority::BelowNormal:
+    percent = 20;
+    break;
+  case ThreadPriority::Normal:
+    percent = 50;
+    break;
+  case ThreadPriority::AboveNormal:
+    percent = 66;
+    break;
+  case ThreadPriority::Highest:
+    percent = 83;
+    break;
+  case ThreadPriority::TimeCritical:
+    percent = 100;
+    break;
+  default:
+    throw std::runtime_error("Attempted to assign an unrecognized thread priority");
+  }
+  param.sched_priority = PTHREAD_MIN_PRIORITY + (percent*(PTHREAD_MAX_PRIORITY - PTHREAD_MIN_PRIORITY) + 50) / 100;
```

Comment:
#include <sched.h>
int sched_get_priority_max(int policy);
int sched_get_priority_min(int policy);
should be used here.